src/playlist/playlistlistview.h

```diff
@@ -24,29 +24,47 @@
 
 #include "config.h"
 
+#include <QBasicTimer>
 #include <QObject>
 #include <QWidget>
 #include <QString>
 
 #include "widgets/autoexpandingtreeview.h"
 
 class QPaintEvent;
+class QDragEnterEvent;
+class QDragLeaveEvent;
+class QDragMoveEvent;
+class QDropEvent;
+class QTimerEvent;
 
 class PlaylistListView : public AutoExpandingTreeView {
   Q_OBJECT
 
  public:
   explicit PlaylistListView(QWidget *parent = nullptr);
 
+  static const int kDragHoverTimeout = 500;
+
   bool ItemsSelected() const;
 
  signals:
   void ItemsSelectedChanged(const bool);
+  void ItemMimeDataDroppedSignal(const QModelIndex &proxy_idx, const QMimeData *q_mimedata);
```

Comment:
No const for q_mimedata

Reply:
are you sure? `const` came from this method https://doc.qt.io/qt-6/qdropevent.html#mimeData

I have to cast it...

Reply:
OK, keep const then.